docs/rule/no-route-action.md

```diff
@@ -0,0 +1,51 @@
+# no-route-action
+
+This rule disallows the usage of `route-action`.
+
+[ember-route-action-helper](https://github.com/DockYard/ember-route-action-helper) is a popular addon that it's only purpose was to avoid creating controllers, something we no longer have a desire to do given the current state of the framework.
+
+Most route actions should either be sent to the controller first or encapsulated within a downstream component instead. We should never be escaping the DDAU hierarchy to lob actions up to the route.
+
+## Examples
+
+This rule **forbids** the following:
+
+```hbs
+{{custom-component onUpdate=(route-action 'updateFoo')}}
+```
+
+```hbs
+{{custom-component onUpdate=(route-action 'updateFoo' 'bar')}}
+```
+
```

Comment:
I like the pattern eslint uses to show good `Examples of incorrect code for this rule` and `Examples of correct code for this rule`.  I suggest using that pattern here.  

Incorrect:
It would be useful to see an example where someone defined an action on the route file and used route-action in the hbs file.

Correct:
Show a controller file, a route file (with no action), and the hbs file.

[example](https://eslint.org/docs/rules/no-empty)

Reply:
Good suggestion.

I didn't want to mix too much JS with this template lint rule. I just added a small JS followed by what is forbidden and allowed. On top of that I added a small migration example to show the whole example.